Batteries/Data/Fin/Lemmas.lean

```diff
@@ -126,31 +133,48 @@ theorem find?_succ {p : Fin (n+1) → Bool} :
   simp only [find?, findSome?_succ, Option.guard]
   split <;> simp [map_findSome?, Function.comp_def, Option.guard]
 
-theorem eq_true_of_find?_eq_some {p : Fin n → Bool} (h : find? p = some i) : p i = true := by
-  match exists_of_findSome?_eq_some h with
-  | ⟨i, hi⟩ =>
-    simp only [Option.guard] at hi
-    split at hi
-    · cases hi; assumption
-    · contradiction
-
-theorem eq_false_of_find?_eq_none {p : Fin n → Bool} (h : find? p = none) (i) : p i = false := by
-  have hi := eq_none_of_findSome?_eq_none h i
-  simp only [Option.guard] at hi
-  split at hi
-  · contradiction
-  · simp [*]
-
-theorem find?_isSome_iff {p : Fin n → Bool} : (find? p).isSome ↔ ∃ i, p i := by
+theorem find?_eq_some_iff {p : Fin n → Bool} :
+    find? p = some i ↔ p i ∧ ∀ j, j < i → ¬ p j := by
+  simp [find?, and_assoc]
+
+theorem find?_isSome_iff {p : Fin n → Bool} :
+    (find? p).isSome ↔ ∃ i, p i ∧ ∀ j < i, ¬ p j := by
```

Comment:
As above, this is not stronger than `(find? p).isSome ↔ ∃ i, p i`.

Reply:
I agree: where is the lemma `∃ i, p i  ↔ ∃ i, p i ∧ ∀ j < i, ¬ p j` that you would need to link them?